Remove user's avatar URL and displayname when deactivate an user with Admin API (#8932)
matrix-org/synapse: docs/admin_api/user_admin_api.rst

- ``deactivated``, optional. If unspecified, deactivation state will be left
  unchanged on existing accounts and set to ``false`` for new accounts.

>   It does not set ``erase``. For details to deactivation see also
>   `Deactivate Account <#deactivate-account>`_.

Reviewer: This sentence is a bit confusing. Maybe the following is clearer?

```suggestion
  A user cannot be erased by deactivating with this API. For details on deactivating users see
  `Deactivate Account <#deactivate-account>`_.
```

The erase parameter is optional and defaults to ``false``.
An empty body may be passed for backwards compatibility.

The following actions are performed when deactivating an user:

- Try to unpind 3PIDs from Identity Server
- Remove all 3PIDs from Homeserver
- Delete any devices and E2E-Keys
- Delete any access tokens

> - Delete passwort hash

Reviewer: ```suggestion
- Delete password hash
```

- Removal from all the rooms the user is a member of
- Delete from user directory
- Reject all pending invites
- Remove all information on the user from the account_validity table

The following actions are additionally performed when deactivating an user and
and you set ``erase`` to ``true``:

- Remove displayname

> - Remove avatar URL

Reviewer: ```suggestion
- Remove avatar URL
- Mark the user as erased
```
